packages/libraries/cli/src/helpers/texture/texture.ts

```diff
@@ -0,0 +1,146 @@
+import { inspect as nodeInspect } from 'node:util';
+import colors from 'colors';
+
+export { colors };
+
+export const space = ' ';
+
+export const indent = space.repeat(3);
+
+export const newline = '\n';
+
+export const header = (value: string) => colors.dim('=== ') + colors.bold(value);
+
+export const plural = (value: unknown[]) => (value.length > 1 ? 's' : '');
+
+export const trimEnd = (value: string) => value.replace(/\s+$/g, '');
+
+/**
+ * Convert quoted text to bolded text. Quotes are stripped.
+ */
+export const boldQuotedWords = (value: string) => {
+  const singleQuotedTextRegex = /'([^']+)'/gim;
+  const doubleQuotedTextRegex = /"([^"]+)"/gim;
+  return value
+    .replace(singleQuotedTextRegex, (_, capturedValue: string) => colors.bold(capturedValue))
+    .replace(doubleQuotedTextRegex, (_, capturedValue: string) => colors.bold(capturedValue));
+};
+
+export const prefixedInspect =
+  (prefix: string) =>
+  (...values: unknown[]) => {
+    const body = values.map(inspect).join(' ');
+    return [prefix, body].join(' ');
+  };
+
+export const inspect = (value: unknown) => {
+  if (typeof value === 'string') {
+    return value;
+  }
+  return nodeInspect(value);
+};
+
+export const success = (...values: unknown[]) =>
+  prefixedInspect(colors.green('✔'))(...values) + newline;
+
+export const failure = (...values: unknown[]) =>
+  prefixedInspect(colors.red('✖'))(...values) + newline;
+
+export const info = (...values: unknown[]) =>
+  prefixedInspect(colors.yellow('ℹ'))(...values) + newline;
+
+export const warning = (...values: unknown[]) =>
+  prefixedInspect(colors.yellow('⚠'))(...values) + newline;
+
+/**
+ * A text builder. Its methods mutate an internal string value.
+ * Useful for quickly building up content.
+ */
+export interface Builder {
```

Comment:
This permits string construction in a way that looks like logging. Especially useful for side-effect style programming e.g. `forEach` loops etc.